core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonConfig.java

```diff
@@ -17,14 +19,19 @@
  * This class is responsible for consuming a reader and producing a tree of config values to be
  * consumed by modules.
  */
-public class MobileCoreJsonParser {
+public class MobileCoreJsonConfig {
```

Comment:
Good catch.  This class should stay MobileCoreJsonParser, but the parse method should return a MobileCoreJsonConfig.  The static parsing methods should stay in the parser class, the parser class should become final, and the config class should be extracted from this class to more clearly separate the concerns